pkg/controller/common/finalizer/finalizers.go

```diff
@@ -0,0 +1,24 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package finalizer
+
+import (
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// RemoveAll removes all existing Finalizers on an Object
+func RemoveAll(c k8s.Client, obj runtime.Object) error {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return err
+	}
+	if len(accessor.GetFinalizers()) == 0 {
+		return nil
+	}
+	accessor.SetFinalizers([]string{})
```

Comment:
We should remove our own finalizers only here (ones that contain `k8s.elastic.co`?), because we don't want to touch user-defined finalizers.